app/ts/utils/uniswap.ts

```diff
@@ -0,0 +1,159 @@
+import { AbiCoder, getCreate2Address, keccak256, solidityPacked } from "ethers"
+import { EthereumAddress } from "./wire-types.js"
+import { Interface } from "ethers"
+import { UNISWAP_V2_FACTORY_ADDRESS, UNISWAP_V3_FACTORY_ADDRESS } from "./constants.js"
+import { addressString } from "./bigint.js"
+
+interface UniswapPools {
+	token0IsQuote: boolean
+	v2: EthereumAddress
+	v3: {
+		'100': EthereumAddress
+		'500': EthereumAddress
+		'3000': EthereumAddress
+		'10000': EthereumAddress
+	}
+}
+
+interface Multicall3Call {
+	target: string
+	allowFailure: boolean
+	callData: string
+}
+
+export function calculateUniswapPools(tokenA: EthereumAddress, tokenB: EthereumAddress): UniswapPools {
```

Comment:
I believe this function only works on mainnet, should we consider renaming this to something like `calculateUniswapPoolsEthereumMainnet` or something?  Perhaps we should have the factory address as a parameter so this function can be reused on any chain?